test/unidirectional-streams.spec.js

```diff
@@ -93,16 +92,16 @@ describe('unidirectional streams', function () {
       await writer.write(buf)
     }
 
-    await new Promise((resolve) => setTimeout(resolve, 500))
+    await new Promise((resolve) => setTimeout(resolve, 2000))
 
     await writer.ready
     await writer.close()
 
     // the remote will close the session cleanly if everything was ok
-    const result = await client.closed
+    await client.closed
 
-    // should receive the default close info
-    expect(result).to.have.property('reason', '')
-    expect(result).to.have.property('closeCode', 0)
+    // should receive the default close info, not true on chromium
```

Comment:
> // should receive the default close info, not true on chromium

If the server receives the wrong bytes, it'll close the session with the `closeCode` `500` and the `reason` `'data did not match'` - by commenting out these assertions we don't guarantee that branch was not hit on the server.

Reply:
> // time out is needed, since it can be received before the read is complete
Ups hard to remember, what I meant. I think it was the following.
Quic does not have head-of-line blocking, all streams and the session have a separate flow. So actually the session close can arrive before the read was complete. (I tracked the events down one by one) And this failed the test a couple of times since the read was intentionally blocked in this test. The delay ensures that the close arrives after the data so that the test can succeed. It is a time issue and did not occur by chance on the node tests.

Reply:
> > // should receive the default close info, not true on chromium
> 
> If the server receives the wrong bytes, it'll close the session with the `closeCode` `500` and the `reason` `'data did not match'` - by commenting out these assertions we don't guarantee that branch was not hit on the server.

That is true. But the result on chromium was different as if it was not passed on, but it may interfere with the other change. I think I changed this first. Anyway, the CI is still failing this test, but I do not get a hint why. On my machine it is only occasionally failing with the connection loss problem, which I have no clue. (In a long-running test I also see quiche black hole detection..., no clue why)

Reply:
> So actually the session close can arrive before the read was complete. (I tracked the events down one by one) 

Maybe I'm misunderstanding something here but are you sure this is possible?  In the test the client sends the bytes and waits for the session to end.  The server receives the bytes and then closes the session, either with an error code if the bytes were wrong or without if the bytes were correct.

As I see it the session cannot be closed before the read has completed because the server waits to read the expected number of bytes, e.g. it completes the read, before it closes the session.

I guess maybe somehow the client might receive the session close before it thinks it's finished writing the data?

> That is true. But the result on chromium was different as if it was not passed on, but it may interfere with the other change.

If we can't rely on the `closeCode`/`reason` fields this sounds like a bug?

> Anyway, the CI is still failing this test, but I do not get a hint why. 

Hmm, yeah - I had this failing reliably the other day, today it passes every time. :shakes fist:.

Reply:
I do not remember everything in detail. It may as well be that the data was received but the fin came later and it was not the session. All I do recall, that it was a weird timing issue that arose since the data transfer was blocked but the closing not.
But anyway, this last unittest tests something internal in the client specific to the node implementation. So I do not think it makes sense at all to use it with chromium as client. 

I am not sure about closeCode, I may test it yourself with logging. But again we are testing Chromiums behavior here.

And yes, this not failing is a problem. On my machine, it really depends on whether or not the Backup process of my harddisk was running. More load more failures.